src/rpc/blockchain.cpp

```diff
@@ -236,7 +225,7 @@ UniValue blockToJSON(const CBlock& block, const CBlockIndex* tip, const CBlockIn
             UniValue objTx(UniValue::VOBJ);
             TxToUniv(*tx, uint256(), objTx, true);
             bool fLocked = isman.IsLocked(tx->GetHash());
-            objTx.pushKV("instantlock", fLocked || chainLock);
+            objTx.pushKV("instantlock", fLocked || result["chainlock"].get_bool());
```

Comment:
instead making OR here, let's add calculating of `instantlock` inside `blockheaderToJSON`, so, users of `blockheaderToJSON` will get `instantlock` too

Reply:
Changing the function return type to `std::pair<UniValue, bool>` or changing the argument list to accept an additional bool for output that outside of this use case would be a discarded value doesn't seem the most elegant to me.

It's guaranteed that `chainlock` will be defined within `result`, it's not efficient but it's less messy imo.

Reply:
> users of blockheaderToJSON will get instantlock too

for sack of unification

Reply:
you can't calculate `instantlock` inside `blockheaderToJSON` - there are no txes in block header

Reply:
ah indeed